ddl/job_table.go

```diff
@@ -460,15 +455,33 @@ func removeDDLReorgHandle(sess *session, job *model.Job, elements []*meta.Elemen
 		return nil
 	}
 	sql := fmt.Sprintf("delete from mysql.tidb_ddl_reorg where job_id = %d", job.ID)
-	_, err := sess.execute(context.Background(), sql, "remove_handle")
-	return err
+	return runInTxn(sess, func(se *session) error {
```

Comment:
runInTxn may wrongly commit the previous session. There is no need to use explicit txn.

Reply:
In newReorgHandler, we need a new session instead of the DDL session.

Reply:
Done, before calling newReorgHandler, it will take out one new session from the session pool and use for the ReorgHandler.